packages/fcl-wc/src/service.js

```diff
@@ -0,0 +1,179 @@
+
+        session = await connectWc(onClose, {
+          client,
+          pairing,
+        })
+      }
+
+      const [namespace, reference, address] = Object.values(session.namespaces)
+        .map(namespace => namespace.accounts)
+        .flat()
+        .filter(account => account.startsWith("flow:"))[0]
+        .split(":")
+
+      const method = service.endpoint
+      const chainId = `${namespace}:${reference}`
+      const addr = address
+      const data = JSON.stringify({...body, addr})
+
+      try {
+        const result = await client.request({
+          topic: session.topic,
+          chainId,
+          request: {
+            method,
+            params: [data],
+          },
+        })
+        onResponse(result)
+      } catch (e) {
+        console.error("Error on WalletConnect request", e)
+        reject(`Declined: Externally Halted`)
+      }
+    })
+  }
+}
+
+async function connectWc(onClose, {client, pairing}) {
+  try {
+    const network = await config.get("flow.network")
+    invariant(
+      network === "mainnet" || network === "testnet",
+      "FCL Configuration value for 'flow.network' is required (testnet || mainnet)"
+    )
+
+    const requiredNamespaces = {
+      flow: {
+        methods: ["flow_authn", "flow_authz", "flow_user_sign"],
+        chains: [`flow:${network}`],
+        events: ["chainChanged", "accountsChanged"],
+      },
+    }
+
+    const {uri, approval} = await client.connect({
+      pairingTopic: pairing?.topic,
+      requiredNamespaces,
+    })
+
+    if (uri) {
+      QRCodeModal.open(uri, () => {
+        console.log("EVENT", "QR Code Modal closed")
+        onClose()
+      })
+    }
+
+    const session = await approval()
+    return session
+  } catch (e) {
+    console.error("Error establishing session", e)
+    throw e
+  } finally {
+    QRCodeModal.close()
+  }
+}
+
+function makeWcServices(client) {
+  const pairings = client.pairing.getAll({active: true})
+  return [
+    {
+      definition: {
+        f_type: "Service",
+        f_vsn: "1.0.0",
+        type: "authn",
+        method: "WC/RPC",
+        uid: "wc#authn",
+        endpoint: "flow_authn",
+        optIn: false,
+        provider: {
+          address: "WalletConnect",
+          name: "WalletConnect",
+          icon: "https://avatars.githubusercontent.com/u/37784886",
+          description: "WalletConnect Generic Provider",
+          website: "https://walletconnect.com/",
+          color: null,
+          supportEmail: null,
+        },
+      },
+      strategy: makeServiceStrategy(client),
+    },
+    ...pairings.map(pairing => {
+      return {
+        definition: {
+          f_type: "Service",
+          f_vsn: "1.0.0",
+          type: "authn",
+          method: "WC/RPC",
+          uid: pairing.topic,
+          endpoint: "flow_authn",
+          optIn: false,
+          provider: {
+            ...pairing,
+            address: pairing.topic,
```

Comment:
Can we create a different field than using this? Otherwise, this makes this field unreliable. Maybe generatedId or something generic?

Reply:
I'm with this. Question is if it requires a version update to the Service. cc/ @JeffreyDoyle 
Should address then be a placeholder as in the generic service for existing pairings?

Reply:
Added `provider.uid` to handle this

Reply:
Yea let's rethink this. We're now seriously using address incorrectly :/ what are we trying to get from having address = paring.topic here?

Reply:
I believe if the provider.address is the same, discovery will dedupe

Reply:
We also need to know the deeplink/universal url for the wallet service